### ISSUE TITLE
Add istio `ServiceEntry` support into networking chart

### ISSUE THREAD

**Issue by zapkub** [argoproj/argo-cd#20270]
# Summary
Currenty there is a partial built-in support for `istio` related resource `VirtualService` in ArgoCD UI. Istio has several networking related resource but this proposal is focusing on adding `ServiceEntry` support

# Motivation
Because networking is really useful for a quick overviewing the consistent of the configuration. I would like to add a missing important istio network component

# Proposal
As like `VirtualService` support. `ServiceEntry` should be add into the network diagram and reference to a workload by `workloadSelector`

TLDR; ServiceEntry works as same as traditional k8s service but with some extra feature

Reference: [Istio ServiceEntry API](https://istio.io/latest/docs/reference/config/networking/service-entry/#ServiceEntry)

### before
<img width="1066" alt="image" src="https://github.com/user-attachments/assets/3001da56-f05d-4b91-8d4b-71de14bef8f9">

### after
<img width="1035" alt="Screenshot 2024-10-08 at 0 15 33" src="https://github.com/user-attachments/assets/fd925565-c9a5-446e-a8b6-ffc8f500f023">
